servicesExternal/web3_single.py

```diff
@@ -182,6 +183,50 @@ def areSameAddressesNoChecksum(self, address1: str, address2: str) -> bool:
             return False
         return address1.toLowerCase() == address2.toLowerCase()
 
+    def keccak(bytes) -> str:
+        return sha3.keccak_256(bytes).digest()
+
+    def message_hash(message) -> str:
+	'''
+	Returns the keccak-256 hash of message, prefixed with the header used by the eth_sign RPC call
+	ethereumjs-util equivalent: https://github.com/ethereumjs/ethereumjs-util/blob/master/docs/index.md#hashpersonalmessage
+	'''
+        if type(message) != bytes:
+	    message = message.encode('utf8')
+        prefix = '\u0019Ethereum Signed Message:\n' + str(len(message))
+        prefix = prefix.encode('utf8')
+        return keccak(prefix + message)
+
+    def legendre_check(r) -> bool:
+        # Legendre symbol check; the secp256k1 library does not seem to do this
+        SECP256K1P = 2**256 - 4294968273
+        xc = r * r * r + 7
+        return pow(xc, (SECP256K1P - 1) // 2, SECP256K1P) == 1
+
+    def ecrecover_to_pub(hash, signature, chaincode = 27) -> str:
```

Comment:
Can you add types to all of your function params (for all functions not just this one)